### PR TITLE
fix starring projects

### PR DESCRIPTION
**Pull request by sandlerr**
The `CurrentProject` concern used in the `StarsController` pulls out the `project_id` param, not the `id` param.

@zendesk/samson @miogalang @ryanseddon

### CODE REVIEW

**Comment by xuebinglee** [comment]
:+1: on :green_apple:

**Comment by miogalang** [comment]
:+1:

**Comment by sandlerr** [comment]
@xuebinglee @miogalang have changed the implementation (ignore the red from CodeClimate please)

**Comment by miogalang** [comment]
:+1:

File: app/controllers/concerns/current_project.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ def current_project
 
   def require_project
     @project = (Project.find_by_param!(params[:project_id]) if params[:project_id])
+    @project ||= (Project.find_by_param!(params[:id]) if params[:id])
```
Comment (sbrnunes) [inline app/controllers/concerns/current_project.rb]: I think loading the project by id as a fallback might be dangerous and lead us to load the wrong project. In a URL using both `project_id` and `id` as path variables, we wouldn't want this to happen.

In another similar situation, we overrided the `require_project` method to match the specific need. See `ProjectsController` for an example of what I mean.

Comment (henders) [inline app/controllers/concerns/current_project.rb]: This might cause problems on the stages controller, as it uses this function as well. I.e. If user tries to access a project page with a bad project_id param, it'll fall back to trying to find a project with the stages' param[:id].

I don't like it, but the solution is probably to do what the LocksController does and override the require_project.

Comment (henders) [inline app/controllers/concerns/current_project.rb]: Ah, @sbrnunes got there before me.

**Comment by msufa** [comment]
I've implemented the fix as @sbrnunes & @henders suggested in #620. I've also added some tests so that we catch similar issues early next time. You can close this PR.

**Comment by henders** [comment]
:-1: as I think #620 fixes some edge conditions, and added tests. :smile_cat:

**Comment by sandlerr** [comment]
Thanks guys
